lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/UpdateTaskHost.cs

```diff
@@ -148,27 +155,80 @@ public byte[] Call(byte[] memento)
                     "Received OperationCanceledException in UpdateTaskHost with message: {0}.",
                     e.Message);
             }
-            catch (IOException e)
+            catch (Exception e)
             {
-                Logger.Log(Level.Error, "Received IOException in UpdateTaskHost with message: {0}.", e.Message);
-                if (!_cancellationSource.IsCancellationRequested)
+                if (e is IOException || e is TcpClientConnectionException || e is RemotingException ||
```

Comment:
The way we are using for catch multiple exceptions is one of the recommended approaches in C# 6.0 "a better idiom to catch multiple exceptions".